PurchasesCoreSwiftTests/Purchasing/ProductsManagerTests.swift

```diff
@@ -91,4 +91,25 @@ class ProductsManagerTests: XCTestCase {
         expect(completionCalled).toEventually(beTrue())
         expect(receivedProducts).to(beEmpty())
     }
+    
+    func testCacheProductCachesCorrectly() {
```

Comment:
is it worth having a test for not calling cache product at all if it's an empty set?

Reply:
good catch, I'll add one

Reply:
re-reading this, I'm not sure what you mean - `cacheProduct` only takes a single product, there's no way to call it with an empty set. 

This particular test works by caching all products, then calling `productsWithIdentifiers` and making sure that _that_ call doesn't trigger an `SKProductsRequest`

Reply:
oof yeah i was referring to the `productsWithIdentifiers` method in `RCPurchases`, sorry. that it wouldn't call `self.productsManager productsWithIdentifiers` if the set was empty.

Reply:
oohhh ok, that makes sense. I added the test to RCPurchases, as well as an extra early return in productsManager + a corresponding test. 

the `ProductsManager` would already not make a request if the set was empty, but it would add a task to its queue, so this early return is a slight optimization of that.